v2/buffer.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
Do we also need to copy the LICENSE file from the Go source code?

Reply:
had a similar question - think we also need to update it if a license header is needed

Reply:
added 👍